pages/1.13/installing/production/system-requirements/zookeeper-requirements/index.md

```diff
@@ -0,0 +1,62 @@
+
+`WARN SyncThread:14  fsync-ing the write ahead log in SyncThread:14 took 14818ms which will adversely effect operation latency. See the ZooKeeper troubleshooting guide`
+
+# Recommendations for a healthy DC/OS cluster
+Because ZooKeeper keeps track of state, it is sensitive to timeouts caused by network latency. If you experience issues with network bandwidth being overloaded or with client sessions that are terminated because network connections are too slow, these issues will also make your DC/OS cluster less reliable.
+
+To ensure you have a healthy DC/OS cluster, you should use the following guidelines to deploy and manage ZooKeeper.
+
+## Monitor memory and swap
+You should verify that Zookeeper has enough heap memory. Insufficient memory allocation can affect ZooKeeper performance particularly during garbage collection. Specific memory requirements, however, vary depending on the number of nodes, clients, and schedulers you have deployed for the cluster, and the overall cluster workload you need to run. 
+
+In most cases, you should also configure your ZooKeeper installation to not allow memory swapping. For ZooKeeper operations to function correctly in a timely way, you should avoid allowing swap space to be used. The maximum heap size you allocate for ZooKeeper should not be larger than the real memory available for to ZooKeeper to use.
+
+One simple way you can monitor the read and write performance of a disk device is by running the following commands to measure server throughput and server latency:
+<p>
+<code>dd if=/dev/zero of=/tmp/test1.img bs=1024 count=1</code><br>
+<code>dd if=/dev/zero of=/tmp/test2.img bs=1024 count=1000</code>
+
+## Isolate master nodes from agent nodes
+
+Isolate master nodes from agent nodes to prevent memory and CPU contention, especially when workloads are started on the agents.
+
+The most common ZooKeeper issues occur when ZooKeeper transaction log files are stored on the same solid state drive (SSD) as other components where ZooKeeper write performance becomes a bottleneck that slows down or delays processing of other worker threads.
+    
+ZooKeeper should not share resources with any other processes or services. ZooKeeper writes new transactions to the log before performing updates or sending a response to the client. Dedicating disk space for the ZooKeeper transaction log directory prevents I/O processing by other applications from overloading the disk.
+    
+You should be sure ZooKeeper directories are configured to use fast disks that can complete synchronization (`fsync`) operations successfully in a timely fashion.
+
+## Provision ZooKeeper directories for master nodes
```

Comment:
I'm not sure this captures the ideas completely correctly.

The `/var/lib/dcos` directory must be on a fast local disk. This is required to ensure that files from several DC/OS components have low latency. However, as the disk is still shared between multiple services, this configuration will reduce contention, but will not prevent seeking.

ZooKeeper can benefit from specifically isolating the ZooKeeper transaction log directory (`/var/lib/dcos/exhibitor/zookeeper/transactions`) on a separate device, separate even from `/var/lib/dcos/exhibitor/zookeeper/snapshot`. In this configuration, we would expect to benefit from the lack of seeking. The difficulty with this advice is that there is a limit on how many fast local disks we can expect a node to provide.

Hence, I suggest reducing this text to something like:

You should put the `/var/lib/dcos` directory for each master node on a separate dedicated disk to reduce latency issues. Using a dedicated partition may not be sufficient. Putting the log on a busy or shared device will adversely affect performance.